CMakeModules/build_clBLAS.cmake

```diff
@@ -21,7 +27,8 @@ ExternalProject_Add(
     INSTALL_DIR "${prefix}"
     UPDATE_COMMAND ""
     DOWNLOAD_NO_PROGRESS 1
-    CONFIGURE_COMMAND ${CMAKE_COMMAND} "-G${CMAKE_GENERATOR}" -Wno-dev <SOURCE_DIR>/src
+    CONFIGURE_COMMAND ${CMAKE_COMMAND} ${extproj_gen_opts}
+      -Wno-dev <SOURCE_DIR>/src
```

Comment:
Can't you just set `-DCMAKE_GENERATOR`? Setting the configure command seems like a hack.

Reply:
`-A` is valid option for `cmake` command and it can't take empty string on non-Windows platform (verified that on Linux). Without forwarding this generator platform for `Visual Studio 16 2019`, cmake is choosing `x86` as platform target even if ArrayFire's platform target is x64. Hence, causing compilation failures due to mismatch in target platforms between third party projects and ArrayFire project. I have already explained this briefly in PR description.

We are conditionally forwarding value of `CMAKE_GENERATOR_PLATFORM` variable to external projects, I don't understand how that is a hack!